scripts/profiler/processed_events.py

```diff
@@ -0,0 +1,58 @@
+from events import EventsData, TrackedEvent
+import logging
+import sys
+
+class ProcessedEvents():
+    def __init__(self):
+        self.raw_data = EventsData([], {})
+        self.tracked_events = []
+
+        self.event_processing_start_id = None
+        self.event_processing_end_id = None
+        self.tracking_execution = True
+
+        self.submit_event = None
+        self.start_event = None
+
+        self.logger = logging.getLogger('Processed Events')
+        self.logger_console = logging.StreamHandler()
+        self.logger.setLevel(logging.WARNING)
+        self.log_format = logging.Formatter(
+            '[%(levelname)s] %(name)s: %(message)s')
+        self.logger_console.setFormatter(self.log_format)
+        self.logger.addHandler(self.logger_console)
+
+    def match_event_processing(self):
+        self.event_processing_start_id = self.raw_data.get_event_type_id(
+            'event_processing_start')
+        self.event_processing_end_id = self.raw_data.get_event_type_id(
+            'event_processing_end')
+
+        if (self.event_processing_start_id is None) or \
+          (self.event_processing_end_id is None):
+            self.tracking_execution = False
+            for ev in self.raw_data.events:
+                self.tracked_events.append(TrackedEvent(ev, None, None))
+            return
+
+        for i in range(0, len(self.raw_data.events)):
+            if self.raw_data.events[i].type_id == self.event_processing_start_id:
+                self.start_event = self.raw_data.events[i]
+                for j in range(i - 1, -1, -1):
```

Comment:
is it possible that you will get to the `j` loop when `i` is `0`? If yes, you will get `-1` index and process some element twice

Reply:
Ok, we may process events starting from i = 1